standard/variables.md

```diff
@@ -901,3 +901,139 @@ variable_reference
+
+There are three different values for *ref-safe-to-escape-scope*: *block in which variable is declared*, *current-method* and *calling-method*. When the *ref-safe-to-escape-scope* is *calling-method*, the variable is *safe-to-return* by reference.
+
+An expression whose type is not a `ref struct` type is *safe-to-return* from the entire enclosing method. Otherwise we refer to the rules below.
+
+### §ref-span-safety-locals Local variable escape scope
+
+For a local variable `l`:
+
+- If `l` is a `ref` variable, its *ref-safe-to-escape-scope* is taken from the *ref-safe-to-escape-scope* of its initializing expression.
+- Otherwise its *ref-safe-to-escape-scope* is the scope in which it was declared.
+
+The *safe-to-escape-scope* of an expression that designates the use of a local variable is determined as follows:
+
+- A local whose type is not a `ref struct` type is *safe-to-return* from the entire enclosing method. Its *safe-to-escape-scope* is the *calling-method*.
+- If the variable is an iteration variable of a `foreach` loop, then the variable's *safe-to-escape-scope* is the same as the *safe-to-escape-scope* of the `foreach` loop's expression.
+- A local of `ref struct` type and uninitialized at the point of declaration is *safe-to-return* from the entire enclosing method. Its *safe-to-escape-scope* is the calling method.
+- Otherwise the variable's type is a `ref struct` type, and the variable's declaration requires an initializer. The variable's *safe-to-escape-scope* is the same as the *safe-to-escape* of its initializer.
+
+### §ref-span-safety-parameters Parameter escape scope
+
+For a formal parameter `p`:
+
+- If `p` is a `ref`, or `in` parameter, its *ref-safe-to-escape-scope* is the *calling-method*. It is *safe-to-return* by ref. If `p` is an `in` parameter, it can't be returned as a writable `ref` but can be returned as `ref readonly`.
+- If `p` is an `out` parameter, its *ref-safe-to-escape-scope* is the *current-method*. It isn't *safe-to-return* by ref.
+- Otherwise, if `p` is the `this` parameter of a struct type, its *ref-safe-to-escape-scope* is the *current-method*.
+- Otherwise, the parameter is a value parameter, and it is *ref-safe-to-escape-scope* is the *current-method*
+
+For an expression that designates the value of a formal parameter `p`, its *safe-to-escape-scope* is the calling method. This applies to the `this` parameter.
+
+### §ref-span-safety-field-reference Field reference escape scope
+
+For a variable designating a reference to a field, `e.F`:
+
+- If `e` is of a reference type, its *ref-safe-to-escape-scope* is the *calling-method*.
+- Otherwise, if `e` is of a value type, its *ref-safe-to-escape-scope* is taken from the *ref-safe-to-escape* of `e`.
```

Comment:
Should this section also include ref fields? If the field `e.F` is a ref field, where `e` is a of `ref struct` type, the _ref-safe-to-escape-scope` of `ref e.F` is the _safe-to-escape_ of `e`.

Reply:
Not yet. The committee is on V7.x, so ref fields aren't supported yet.